File: Interview Preparation Kit/Recursion and Backtracking/FibonacciNumbers.c
// https://www.hackerrank.com/challenges/ctci-fibonacci-numbers/problem
#include <stdio.h>

int fibonacci(int n) {
    if (n < 2) {
        return n;
    }

    int a = 1;
    int b = 1;
    int current;

    for (int i = 2; i < n; i++) {
        current = a + b;
        a = b;
        b = current;
    }
    
    return b;
}

int main() {
    int n;
    scanf("%d", &n);
    printf("%d", fibonacci(n));
    return 0;
}

File: Interview Preparation Kit/Recursion and Backtracking/FibonacciNumbers.go
// https://www.hackerrank.com/challenges/ctci-fibonacci-numbers/problem
package main

import "fmt"

func fibonacci(n int) int {
    if n < 2 {
        return n
    }

    a, b, current := 1, 1, 0

    for i := 2; i < n; i++ {
        current = a + b
        a = b
        b = current
    }

    return b
}

func main() {
    var n int
    fmt.Scanf("%d\n", &n)
    fmt.Println(fibonacci(n))
}
